### PR TITLE
fix: rename sourcemap flag to devtool

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

fix/refactor

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes

**Summary**
- Sorucemap flag is misleading, rename it to devtool to align with the config option

**Does this PR introduce a breaking change?**
Yes

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -194,8 +194,8 @@ module.exports = {
             link: 'https://webpack.js.org/concepts/hot-module-replacement/',
         },
         {
-            name: 'sourcemap',
-            usage: '--sourcemap <sourcemap | eval>',
+            name: 'devtool',
+            usage: '--devtool <source-map | eval>',
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/utils/cli-flags.js:198]: need fix it, because we have more values here

Comment (anshumanv) [reply packages/webpack-cli/lib/utils/cli-flags.js]: done @evilebottnawi 👍

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 2]
```diff
@@ -194,8 +194,8 @@ module.exports = {
             link: 'https://webpack.js.org/concepts/hot-module-replacement/',
         },
         {
-            name: 'sourcemap',
-            usage: '--sourcemap <sourcemap | eval>',
+            name: 'devtool',
+            usage: '--devtool <value>',
             type: String,
             alias: 's',
```
Comment (snitin315) [inline packages/webpack-cli/lib/utils/cli-flags.js:200]:
```suggestion
            alias: 'd',
```

File: packages/webpack-cli/lib/groups/BasicGroup.js [hunk 3]
```diff
@@ -30,9 +30,8 @@ class BasicGroup extends GroupHelper {
             // const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/BasicGroup.js:30]: We need to remove these lines, let's do it in other PR

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/BasicGroup.js:30]: sure yes

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/BasicGroup.js:30]: @evilebottnawi already handled here - https://github.com/webpack/webpack-cli/pull/1361

Will complete it later and refactor to its own package

**Pull request merged**